lnwire/node_announcement_2.go

```diff
@@ -0,0 +1,506 @@
+package lnwire
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"image/color"
+	"io"
+	"net"
+
+	"github.com/lightningnetwork/lnd/tlv"
+	"github.com/lightningnetwork/lnd/tor"
+)
+
+// NodeAnnouncement2 message is used to announce the presence of a Lightning
+// node and also to signal that the node is accepting incoming connections.
+// Each NodeAnnouncement2 authenticating the advertised information within the
+// announcement via a signature using the advertised node pubkey.
+type NodeAnnouncement2 struct {
+	// Features is the feature vector that encodes the features supported
+	// by the target node.
+	Features tlv.RecordT[tlv.TlvType0, RawFeatureVector]
+
+	// Color is an optional field used to customize a node's appearance in
+	// maps and graphs.
+	Color tlv.OptionalRecordT[tlv.TlvType1, Color]
+
+	// BlockHeight allows ordering in the case of multiple announcements. We
+	// should ignore the message if block height is not greater than the
+	// last-received. The block height must always be greater or equal to
+	// the block height that the channel funding transaction was confirmed
+	// in.
+	BlockHeight tlv.RecordT[tlv.TlvType2, uint32]
+
+	// Alias is used to customize their node's appearance in maps and
+	// graphs.
+	Alias tlv.OptionalRecordT[tlv.TlvType3, []byte]
```

Comment:
I think this could use a separate type, in the spec it says it's [32]byte of utf8s